packages/core/src/Dialog/Dialog.styled.tsx

```diff
@@ -0,0 +1,169 @@
+import type { IDialogProps } from '..'
+import { Box, CloseButton, Grid, Text, ThemeProvider } from '..'
+
+import * as Dialog from '@radix-ui/react-dialog'
+import { VisuallyHidden } from '@radix-ui/react-visually-hidden'
+import themeGet from '@styled-system/theme-get'
+import { motion } from 'framer-motion'
+import React from 'react'
+import styled from 'styled-components'
+
+export const DialogSizes = ['sm', 'md', 'lg', 'xl', 'full'] as const
+export type DialogSize = typeof DialogSizes[number]
+
+type SizeStyle = {
+  width?: string
+  height?: string
+  borderRadius?: string
+}
+
+const sizeStyles: Record<DialogSize, SizeStyle> = {
+  sm: {
+    width: 'calc(min(400px, 100vw) - 2 * 24px)',
+    borderRadius: '24px',
+  },
+  md: {
+    width: 'calc(min(640px, 100vw) - 2 * 24px)',
+    borderRadius: '24px',
+  },
+  lg: {
+    width: 'calc(min(960px, 100vw) - 2 * 24px)',
+    borderRadius: '24px',
+  },
+  xl: {
+    width: 'calc(min(1280px, 100vw) - 2 * 24px)',
+    height: 'calc(100% - 2 * 24px)',
+    borderRadius: '24px',
+  },
+  full: {
+    width: '100%',
+    height: '100%',
+  },
+} as const
+
+const FloatingCloseButton: typeof CloseButton = styled(CloseButton)`
+  position: absolute;
+  top: ${(props) => -((props.dialogSize === 'full' ? 0 : 16) + 10)}px;
+  right: ${(props) => -((props.dialogSize === 'full' ? 0 : 16) + 10)}px;
+  margin: 16px;
+  padding: 8px;
+  background-color: white;
+  box-shadow: ${themeGet('shadows.overlay-md')};
+  &:hover {
+    background-color: white;
+  }
+`
+
+const DialogOverlayWrapper = styled(motion.div)`
+  position: fixed;
+  inset: 0;
+  display: grid;
+  overflow-y: auto;
+  cursor: ${(props: IDialogProps) => (props.scrimDismiss ? 'pointer' : 'default')};
+  place-items: ${(props: IDialogProps) => (props.sheet ? 'end center' : 'center')};
+  background-color: ${(props: IDialogProps) => `rgba(0, 0, 0, ${props.scrimOpacity})`};
+`
+
+const DialogContentWrapper = styled(motion.div)`
+  position: relative;
+  cursor: initial;
+  background-color: white;
```

Comment:
there's some colors and other things that should be coming from the theme that might need a little cleanup